include/deal.II/dofs/dof_vector.h

```diff
@@ -0,0 +1,243 @@
+ * without their DoFHandler. Nevertheless, deal.II has assumed for
+ * decades, that there is a single DoFHandler in a program and that
+ * all vectors refer to it. This class encapsulates both, to begin
+ * with in a minimally intrusive way, giving access to both
+ * structures.
+ *
+ * @todo Ensure that both objects are synchronized, that is, the
+ * DoFHandler is not changes without the vector being resized.
+ */
+template <class DH, class VECTOR=Vector<double> >
+class DoFVector :  public Subscriptor
+{
+public:
+  /// @name Constructors and initialization
+  ///@{
+  /**
+   * Construct an object with a DoFHandler and borrow a vector from
+   * GrowingVectorMemory for its data. This vector is then owned by
+   * this object and released when the object itself expires.
+   */
+  DoFVector (const DH &dh, const ConstraintMatrix &constraints);
+
+  /**
+   * Construct an object with a DoFHandler but without constraints and
+   * borrow a vector from GrowingVectorMemory for its data. This
+   * vector is then owned by this object and released when the object
+   * itself expires.
+   *
+   * @note This constructor uses the ConstraintMatrix
+   * #no_constraints. The created object thus can output vectors in
+   * distributed format, but writing into vectors will most likely
+   * fail without using a constraint matrix from somewhere else.
+   */
+  DoFVector (const DH &dh);
+
+  /**
+   * Construct an object with a DoFHandler and borrow a vector from
+   * GrowingVectorMemory for its data. This vector is not owned by the
+   * object and read-access only.
+   *
+   * @note An object created this way is effectively constant. In
+   * order to avoid obscure access errors to the data vector, it
+   * should also be declared `const`.
+   */
+  DoFVector (const DH &dh, const ConstraintMatrix &constraints, const VECTOR &v);
+
+  /**
+   * Construct an object with a DoFHandler but without constraints and
+   * borrow a vector from GrowingVectorMemory for its data. This
+   * vector is not owned by the object and read-access only.
+   *
+   * @note An object created this way is effectively constant. In
+   * order to avoid access errors to the data vector, it should also
+   * be declared `const`. Also, vectors are expected to be in
+   * distributed format, since the information on constraints is
+   * missing.
+   */
+  DoFVector (const DH &dh, const VECTOR &v);
+
+  /**
+   * Destructor, deleting own data.
+   */
+  ~DoFVector ();
+
+  /**
+   * Modify the vector dimensions to match the current state of the
+   * dof handler.
+   */
+  void sync ();
+
+  ///@}
+
+  /// @name Access to members
+  ///@{
+  /**
+   * Access to the DoFHandler
+   */
+  const DH &dof_handler () const;
+
+  /**
+   * Read-only access to the data vector
+   */
+  const VECTOR &data () const;
+
+  /**
+   * Read-write access to the data vector.
+   *
+   * Read-write access is only possible, if the vector is owned by the
```

Comment:
But then I think I still don't understand what the whole thing is there for. If the intent is to create a DoFVector every time one calls a library function, then one still has to list DoFHandler, ConstraintMatrix, and vector. In that case, there is little gained -- just more to write. On the other hand, if this class is intended as a way for people to only store a DoFVector, rather than a plain vector, then it is necessary that they be able to write to it.